openapi.yaml

```diff
@@ -0,0 +1,1607 @@
+  /pending-surveys/{batch-no}:
+    get:
+      tags:
+        - misc
+      summary: Get pending survey transfers or shares by batch number
+      description: Get pending survey transfers or shares by batch number
+      parameters:
+        - name: batch-no
+          in: path
+          required: true
+          description: The batch number to retrieve
+          schema:
+            type: integer
+      responses:
+        200:
+          description: The pending survey transfers or shares have been retrieved
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/PendingSurvey'
+        404:
+          description: The batch number does not exist
+    delete:
+      tags:
+        - misc
+      summary: Delete pending survey transfers or shares by batch number
+      description: Delete pending survey transfers or shares by batch number
+      parameters:
+        - name: batch-no
+          in: path
+          required: true
+          description: The batch number to delete
+          schema:
+            type: integer
+      responses:
+        202:
+          description: The pending survey transfers or shares have been deleted
+        404:
+          description: The batch number does not exist
+        500:
+          description: A database error occured during the deletion
+  /pending-surveys/originator/{originator-respondent-id}:
+    get:
+      tags:
+        - misc
+      summary: Get pending survey transfers or shares by the respondent ID of the respondent who shared/transferred it away
+      description: Get pending survey transfers or shares by the respondent ID of the respondent who shared/transferred it away
+      parameters:
+        - name: originator-respondent-id
+          in: path
+          required: true
+          description: The UUID of the respondent who shared/transferred the survey
+          schema:
+            type: string
+            format: uuid
+      responses:
+        200:
+          description: The pending survey transfers or shares have been retrieved
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/PendingSurvey'
+  /pending-surveys/resend-email:
+    post:
+      tags:
+        - misc
+      summary: Resend the pending survey transfer/share email
+      description: Resend the pending survey transfer/share email
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                batch_number:
+                  type: integer
+                  description: The batch number of the survey transfer or share
+      responses:
+        201:
+          description: The pending survey transfer/share email has been resent
+        400:
```

Comment:
respondent not uuid

Reply:
This shouldn't be possible even though it's covered for - the API user only provides batch_number so it would require the ID in the database to already not be a UUID